src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx

```diff
@@ -17,6 +21,23 @@ const Stack = createStackNavigator<OnboardingModalNavigatorParamList>();
 function OnboardingModalNavigator() {
     const styles = useThemeStyles();
     const {shouldUseNarrowLayout} = useOnboardingLayout();
+    const [hasCompletedGuidedSetupFlow] = useOnyx(ONYXKEYS.NVP_ONBOARDING, {selector: (onboarding) => onboarding?.hasCompletedGuidedSetupFlow ?? true});
```

Comment:
I see you have changed it here to work with the type change, however, is removing the empty array type compatible with all the other places where we use this nvp?

Reply:
@mountiny 
I checked every place where `ONYXKEYS.NVP_ONBOARDING` is used in app and it does not need to be an array. Even though Backend returns empty array for old accounts, `[].hasCompletedGuidedSetupFlow` is `undefined` so this should work fine.

Let me know your thoughts on this.
Thanks.

Reply:
@skyweb331 I am still not sure if we should remove the empty array type, given this is a valid and expected value for that nvp, I am worried that if people will forget it can be empty array for old accounts, they might make some changes that wont treat the case safely.

Could you please try to put that type back? thanks

Reply:
@mountiny I reverted it. And also added comment why it could be an array.